mesonbuild/compilers/mixins/apple.py

```diff
@@ -59,3 +59,22 @@ def openmp_link_flags(self, env: Environment) -> T.List[str]:
     def get_prelink_args(self, prelink_name: str, obj_list: T.List[str]) -> T.List[str]:
         # The objects are prelinked through the compiler, which injects -lSystem
         return ['-nostdlib', '-r', '-o', prelink_name] + obj_list
+
+
+class AppleCStdsMixin(Compiler):
+
+    """Provide version overrides for the Apple Compilers."""
+
+    _C17_VERSION = '>=10.0.0'
+    _C18_VERSION = '>=11.0.0'
+    _C2X_VERSION = '>=11.0.0'
+
+
+class AppleCPPStdsMixin(Compiler):
+
+    """Provide version overrides for the Apple C++ Compilers."""
+
+    _CPP23_VERSION = '>=13.0.0'
+    # TODO: We don't know which XCode version will include LLVM 17 yet, so
+    # use something absurd.
+    _CPP26_VERSION = '>=99.0.0'
```

Comment:
I don't really like this, but I don't know Apple ecosystem; therefore I cannot suggest anything meaningful...

Reply:
I don't like it either, fortunately I didn't write it, just moved it :)

Reply:
Is #14075 related to this?

Reply:
the mechanical fix for that issues is to change this line you're pointing out.

This series would make that fix apply to both the C++ and ObjC++ Compilers at the same time with the one line change.

Reply:
It's now been changed, so unlikable lines no longer have to be moved.

Reply:
Thanks for fixing that, I've rebased on current master, so this is now resolved.